lib/constellation.rb

```diff
@@ -1,58 +1,31 @@
-require 'net/http'
+require "graphql/client"
+require "graphql/client/http"
 module Constellation
-  class HttpException < StandardError
+  class GraphQLException < StandardError
   end
-  HTTP_ERRORS = [
-    Errno::ETIMEDOUT,
-    Timeout::Error,
-    Net::OpenTimeout,
-    Net::ReadTimeout,
-    EOFError,
-    Errno::ECONNABORTED,
-    Errno::ECONNREFUSED,
-    Errno::ECONNRESET,
-    Errno::EHOSTDOWN,
-    Errno::EHOSTUNREACH,
-    Errno::EINVAL,
-    Errno::ENETUNREACH,
-    SocketError,
-    OpenSSL::SSL::SSLError,
-    Net::HTTPBadResponse,
-    Net::HTTPHeaderSyntaxError,
-    Net::ProtocolError,
-    Zlib::GzipFile::Error,
-  ]
 
-  def self.create_rsvp!(rsvp_params)
-    return unless constellation_enabled?
-    uri = URI.parse("#{app}/rsvps")
-    http = Net::HTTP.new(uri.host, uri.port)
-    http.use_ssl = true
-    req = Net::HTTP::Post.new(uri.path, headers)
-    req.body = { rsvp: rsvp_params.merge(total_count: true) }.to_json
-    response = http.request(req)
-    raise Constellation::HttpException, response.body unless response.code == '201'
-    return JSON.parse(response.body), response['X-Total-Count'].to_i
-  rescue *HTTP_ERRORS => e
-    raise Constellation::HttpException, e.message
-  end
-
-  def self.constellation_enabled?
-    app && token
+  HTTP = GraphQL::Client::HTTP.new("#{RsvpRails.config[:constellation_url]}/api/graphql/") do
```

Comment:
This was sort of awkward, the gem insists that these exist as constants (schema fetched on app load).

Also, the way you pass in `def headers` is bizarre, was hard to figure out.

Reply:
I guess the rationale is to do this on app load to fail earlier (when the schema and all queries are known) rather than later (at runtime). I think that’s a good thing.

I guess they modeled extending the client after how associations in ActiveRecord can het extended? That is, they yield i the context of a `module_eval` and extend the client with that module. Were there docs for this?